src/hotspot/share/prims/jvmtiGetLoadedClasses.cpp

```diff
@@ -102,10 +103,6 @@ JvmtiGetLoadedClasses::getLoadedClasses(JvmtiEnv *env, jint* classCountPtr, jcla
 
   LoadedClassesClosure closure(env, false);
   {
-    // To get a consistent list of classes we need MultiArray_lock to ensure
-    // array classes aren't created.
-    MutexLocker ma(MultiArray_lock);
```

Comment:
Unclear why this is no longer the case.

Reply:
CLD::_klasses list is iterated through lock free with atomics. Older code used to iterate through the system dictionary which only had InstanceKlasses and then used to iterate through the arrays in InstanceKlass::_array_klasses.  Which required the MultiArray_lock protecting it.  There used to be an array_klasses_do() function.   Now all the klasses are in the CLD::_klasses list and traversed atomically.